### DISCUSSION TITLE
Building from source but having ‘CUDA_CALL’ must be available [-fpermissive] error

### DISCUSSION THREAD

**Discussion by androuino** [apache/incubator-mxnet#19536] [Q&A] [answered]
## Description
I was trying to build mxnet from source and my computer's spec is this:
Ubuntu 18.04
Cuda9.0 = V9.0.176
g++-6
gcc-6

### Error Message
`make: cc: Command not found
make: cc: Command not found
Makefile:264: WARNING: Significant performance increases can be achieved by installing and enabling gperftools or jemalloc development packages
INFO: nvcc was not found on your path
INFO: Using /usr/local/cuda/bin/nvcc as nvcc path
Running CUDA_ARCH: -gencode arch=compute_30,code=sm_30 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_70,code=[sm_70,compute_70] --fatbin-options -compress-all
rm -rf python/mxnet/*/*.so python/mxnet/*/*.cpp
rm -f build/tests/cpp/mxnet_unit_tests  build/tests/cpp/test_main.o  build/tests/cpp/engine/threaded_engine_test.o  build/tests/cpp/storage/storage_test.o  build/tests/cpp/operator/slice_channel_perf.o  build/tests/cpp/operator/runner/core_op_runner_test.o  build/tests/cpp/operator/dropout_perf.o  build/tests/cpp/operator/krprod_test.o  build/tests/cpp/operator/activation_perf.o  build/tests/cpp/operator/coreop_perf.o  build/tests/cpp/operator/fully_conn_perf.o  build/tests/cpp/operator/batchnorm_test.o  build/tests/cpp/operator/mkldnn.o  build/tests/cpp/operator/tune/operator_tune_test.o  build/tests/cpp/misc/memory_test.o
rm -f -r build lib bin *~ */*~ */*/*~ */*/*/*~ R-package/NAMESPACE R-package/man R-package/R/mxnet_generated.R \
	R-package/inst R-package/src/image_recordio.h R-package/src/*.o R-package/src/*.so mxnet_*.tar.gz \
	3rdparty/mkldnn/install/*
cd /home/wadachi/incubator-mxnet/3rdparty/dmlc-core; make clean; cd -
make[1]: Entering directory '/home/wadachi/incubator-mxnet/3rdparty/dmlc-core'
rm -f line_split.o indexed_recordio_split.o recordio_split.o input_split_base.o io.o filesys.o local_filesys.o data.o recordio.o config.o  libdmlc.a test/filesys_test test/dataiter_test test/iostream_test test/recordio_test test/split_read_test test/stream_read_test test/split_test test/libsvm_parser_test test/libfm_parser_test test/split_repeat_read_test test/strtonum_test test/logging_test test/parameter_test test/registry_test test/csv_parser_test test/unittest/dmlc_unittest test/unittest/unittest_env.o test/unittest/unittest_thread_group.o test/unittest/unittest_json.o test/unittest/unittest_threaditer.o test/unittest/unittest_optional.o test/unittest/unittest_parser.o test/unittest/unittest_main.o test/unittest/unittest_any.o test/unittest/unittest_array_view.o test/unittest/unittest_lockfree.o test/unittest/unittest_param.o test/unittest/unittest_threaditer_exc_handling.o test/unittest/unittest_serializer.o test/unittest/unittest_logging.o test/unittest/unittest_config.o *~ src/*~ src/*/*~ include/dmlc/*~ test/*~
make[1]: Leaving directory '/home/wadachi/incubator-mxnet/3rdparty/dmlc-core'
/home/wadachi/incubator-mxnet
cd /home/wadachi/incubator-mxnet/3rdparty/ps-lite; make clean; cd -
make[1]: Entering directory '/home/wadachi/incubator-mxnet/3rdparty/ps-lite'
rm -rf build  tests/test_connection  tests/test_simple_app  tests/test_kv_app_multi_workers  tests/test_kv_app tests/*.d
find src -name "*.pb.[ch]*" -delete
make[1]: Leaving directory '/home/wadachi/incubator-mxnet/3rdparty/ps-lite'
/home/wadachi/incubator-mxnet
cd /home/wadachi/incubator-mxnet/3rdparty/nnvm; make clean; cd -
make[1]: Entering directory '/home/wadachi/incubator-mxnet/3rdparty/nnvm'
rm -f -rf build lib bin *~ */*~ */*/*~ */*/*/*~ */*.o */*/*.o */*/*/*.o cli_test
make[1]: Leaving directory '/home/wadachi/incubator-mxnet/3rdparty/nnvm'
/home/wadachi/incubator-mxnet
cd /home/wadachi/incubator-mxnet/amalgamation; make clean; cd -
make[1]: Entering directory '/home/wadachi/incubator-mxnet/amalgamation'
rm -f *.d *.o *.so *.a *.js *.js.mem mxnet_predict-all.cc nnvm.cc
make[1]: Leaving directory '/home/wadachi/incubator-mxnet/amalgamation'
/home/wadachi/incubator-mxnet
make: cc: Command not found
make: cc: Command not found
Makefile:264: WARNING: Significant performance increases can be achieved by installing and enabling gperftools or jemalloc development packages
INFO: nvcc was not found on your path
INFO: Using /usr/local/cuda/bin/nvcc as nvcc path
Running CUDA_ARCH: -gencode arch=compute_30,code=sm_30 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_70,code=[sm_70,compute_70] --fatbin-options -compress-all
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/mkldnn/mkldnn_convolution.cc -o build/src/operator/nn/mkldnn/mkldnn_convolution.o
In file included from src/operator/nn/mkldnn/../../linalg.h:31:0,
                 from src/operator/nn/mkldnn/../convolution-inl.h:44,
                 from src/operator/nn/mkldnn/mkldnn_convolution.cc:26:
src/operator/nn/mkldnn/../.././c_lapack_api.h:330:25: note: #pragma message: Warning: lapack usage not enabled, linalg-operators will not be available. Ensure that lapack library is installed and build with USE_LAPACK=1 to get lapack functionalities.
      " functionalities.")
                         ^
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/mkldnn/mkldnn_concat.cc -o build/src/operator/nn/mkldnn/mkldnn_concat.o
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/mkldnn/mkldnn_base.cc -o build/src/operator/nn/mkldnn/mkldnn_base.o
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/mkldnn/mkldnn_act.cc -o build/src/operator/nn/mkldnn/mkldnn_act.o
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/mkldnn/mkldnn_softmax.cc -o build/src/operator/nn/mkldnn/mkldnn_softmax.o
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/mkldnn/mkldnn_deconvolution.cc -o build/src/operator/nn/mkldnn/mkldnn_deconvolution.o
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/mkldnn/mkldnn_copy.cc -o build/src/operator/nn/mkldnn/mkldnn_copy.o
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/mkldnn/mkldnn_fully_connected.cc -o build/src/operator/nn/mkldnn/mkldnn_fully_connected.o
In file included from src/operator/nn/mkldnn/../../linalg.h:31:0,
                 from src/operator/nn/mkldnn/../fully_connected-inl.h:37,
                 from src/operator/nn/mkldnn/mkldnn_fully_connected.cc:26:
src/operator/nn/mkldnn/../.././c_lapack_api.h:330:25: note: #pragma message: Warning: lapack usage not enabled, linalg-operators will not be available. Ensure that lapack library is installed and build with USE_LAPACK=1 to get lapack functionalities.
      " functionalities.")
                         ^
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/mkldnn/mkldnn_pooling.cc -o build/src/operator/nn/mkldnn/mkldnn_pooling.o
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/mkldnn/mkldnn_sum.cc -o build/src/operator/nn/mkldnn/mkldnn_sum.o
g++ -std=c++11 -c -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/home/wadachi/incubator-mxnet/3rdparty/mshadow/ -I/home/wadachi/incubator-mxnet/3rdparty/dmlc-core/include -fPIC -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/include -I/home/wadachi/incubator-mxnet/3rdparty/dlpack/include -I/home/wadachi/incubator-mxnet/3rdparty/nnvm/tvm/include -Iinclude -funroll-loops -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_F16C=0 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -DMSHADOW_USE_CUDNN=1  -I/home/wadachi/incubator-mxnet/3rdparty/cub -DMXNET_USE_NCCL=0 -DMXNET_USE_LIBJPEG_TURBO=0 -MMD -c src/operator/nn/cudnn/cudnn_algoreg.cc -o build/src/operator/nn/cudnn/cudnn_algoreg.o
In file included from include/mxnet/./resource.h:31:0,
                 from include/mxnet/operator.h:39,
                 from src/operator/nn/cudnn/./../convolution-inl.h:33,
                 from src/operator/nn/cudnn/./cudnn_algoreg-inl.h:34,
                 from src/operator/nn/cudnn/cudnn_algoreg.cc:26:
include/mxnet/./../../src/common/random_generator.h: In static member function ‘static void mxnet::common::random::RandGenerator<mshadow::gpu, DType>::AllocState(mxnet::common::random::RandGenerator<mshadow::gpu, DType>*)’:
include/mxnet/./../../src/common/random_generator.h:154:5: error: there are no arguments to ‘CUDA_CALL’ that depend on a template parameter, so a declaration of ‘CUDA_CALL’ must be available [-fpermissive]
     CUDA_CALL(cudaMalloc(&inst->states_,
     ^~~~~~~~~
include/mxnet/./../../src/common/random_generator.h:154:5: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
include/mxnet/./../../src/common/random_generator.h: In static member function ‘static void mxnet::common::random::RandGenerator<mshadow::gpu, DType>::FreeState(mxnet::common::random::RandGenerator<mshadow::gpu, DType>*)’:
include/mxnet/./../../src/common/random_generator.h:159:5: error: there are no arguments to ‘CUDA_CALL’ that depend on a template parameter, so a declaration of ‘CUDA_CALL’ must be available [-fpermissive]
     CUDA_CALL(cudaFree(inst->states_));
     ^~~~~~~~~
In file included from src/operator/nn/cudnn/./../../linalg.h:31:0,
                 from src/operator/nn/cudnn/./../convolution-inl.h:44,
                 from src/operator/nn/cudnn/./cudnn_algoreg-inl.h:34,
                 from src/operator/nn/cudnn/cudnn_algoreg.cc:26:
src/operator/nn/cudnn/./../.././c_lapack_api.h: At global scope:
src/operator/nn/cudnn/./../.././c_lapack_api.h:330:25: note: #pragma message: Warning: lapack usage not enabled, linalg-operators will not be available. Ensure that lapack library is installed and build with USE_LAPACK=1 to get lapack functionalities.
      " functionalities.")
                         ^
Makefile:431: recipe for target 'build/src/operator/nn/cudnn/cudnn_algoreg.o' failed
make: *** [build/src/operator/nn/cudnn/cudnn_algoreg.o] Error 1`

## To Reproduce
clone the repository and make som changes to the lines in make/config.mk file.
`export CC = gcc-6`
`export CXX = g++-6`
`export NVCC = nvcc`
run `make clean && make`

### Steps to reproduce
(Paste the commands you ran that produced the error.)

1. cd to the incubator-mxnet director
2. make some changes to the lines in make/config.mk file
`export CC = gcc-6`
`export CXX = g++-6`
`export NVCC = nvcc`
3. run `make clean && make`

## What have you tried to solve it?

1. change the line `ADD_CFLAGS = -fpermissive`
2. run the `make clean && make` again but having the same error

## Environment

***We recommend using our script for collecting the diagnostic information with the following command***
`curl --retry 10 -s https://raw.githubusercontent.com/apache/incubator-mxnet/master/tools/diagnose.py | python3`

<details>
<summary>Environment Information</summary>

```
----------Python Info----------
Version      : 3.6.9
Compiler     : GCC 8.4.0
Build        : ('default', 'Oct  8 2020 12:12:24')
Arch         : ('64bit', 'ELF')
------------Pip Info-----------
Version      : 9.0.1
Directory    : /usr/lib/python3/dist-packages/pip
----------MXNet Info-----------
An error occured trying to import mxnet.
This is very likely due to missing missing or incompatible library files.
Traceback (most recent call last):
  File "<stdin>", line 96, in check_mxnet
AttributeError: module 'mxnet' has no attribute '__version__'

----------System Info----------
Platform     : Linux-4.15.0-123-generic-x86_64-with-Ubuntu-18.04-bionic
system       : Linux
node         : nvidia-server
release      : 4.15.0-123-generic
version      : #126-Ubuntu SMP Wed Oct 21 09:40:11 UTC 2020
----------Hardware Info----------
machine      : x86_64
processor    : x86_64
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              2
On-line CPU(s) list: 0,1
Thread(s) per core:  1
Core(s) per socket:  2
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               23
Model name:          Intel(R) Core(TM)2 Duo CPU     E8400  @ 3.00GHz
Stepping:            10
CPU MHz:             2026.542
CPU max MHz:         3003.0000
CPU min MHz:         2003.0000
BogoMIPS:            5984.78
Virtualization:      VT-x
L1d cache:           32K
L1i cache:           32K
L2 cache:            6144K
NUMA node0 CPU(s):   0,1
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl cpuid aperfmperf pni dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm sse4_1 xsave lahf_lm pti tpr_shadow vnmi flexpriority dtherm
----------Network Test----------
Setting timeout: 10
Timing for MXNet: https://github.com/apache/incubator-mxnet, DNS: 0.0111 sec, LOAD: 0.7182 sec.
Timing for Gluon Tutorial(en): http://gluon.mxnet.io, DNS: 0.1118 sec, LOAD: 0.1908 sec.
Error open Gluon Tutorial(cn): https://zh.gluon.ai, <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)>, DNS finished in 0.14885759353637695 sec.
Timing for FashionMNIST: https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/fashion-mnist/train-labels-idx1-ubyte.gz, DNS: 0.0329 sec, LOAD: 0.4919 sec.
Timing for PYPI: https://pypi.python.org/pypi/pip, DNS: 0.0083 sec, LOAD: 0.7822 sec.
Error open Conda: https://repo.continuum.io/pkgs/free/, HTTP Error 403: Forbidden, DNS finished in 0.02862691879272461 sec.
----------Environment----------
```

</details>

Appreciate your help. Thank you very much.

**Comment by leezu** [comment, marked as answer]
Which version of mxnet are you building? If it's a recent version, you can try to use the cmake build instead

**Comment by androuino** [comment]
after cloning it, I didn't check out to any branch so I guess it's the latest version. I will try with cmake. Thanks.
